src/App.tsx

```diff
@@ -1,54 +1,97 @@
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handlSelectedPerson = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setListIsVisible(false);
+  };
+
+  const people = useMemo(() => {
+    return [...peopleFromServer].filter(person => {
+      const name = person.name.toLowerCase();
+      const compareQuery = applieQuery.toLowerCase().trim();
+
+      return name.includes(compareQuery);
+    });
+  }, [applieQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={cn('dropdown', { 'is-active': listIsVisible })}>
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQueryChange}
+            onFocus={() => setListIsVisible(true)}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {people.length === 0
+              ? (
+                <div className="dropdown-item">
+                  <p className="has-text-primary-dark">
+                    No matching suggestions
+                  </p>
+                </div>
+              )
+              : people.map(person => (
+                <div
+                  className="dropdown-item"
+                  key={person.slug}
+                  onClick={() => handlSelectedPerson(person)}
+                  onKeyDown={(event) => {
+                    if (event.key === 'Enter') {
+                      setSelectedPerson(person);
+                    }
+                  }}
```

Comment:
looks like redundatn event handler. 
it is should be in focus some item to work this handler.